tests/test_build.py

```diff
@@ -88,8 +88,8 @@ def dummy_getstatusoutput(x):
       'pip --disable-pip-version-check show alibuild | grep -e "^Version:" | sed -e \'s/.* //\'': (0, "v1.5.0"),
       'which pigz': (1, ""),
       'tar --ignore-failed-read -cvvf /dev/null /dev/zero': (0, ""),
-      'git ls-remote --heads https://github.com/root-mirror/root': (0, "87b87c4322d2a3fad315c919cb2e2dd73f2154dc\trefs/heads/master\nf7b336611753f1f4aaa94222b0d620748ae230c0\trefs/heads/v6-08-00-patches"),
-      'git ls-remote --heads https://github.com/star-externals/zlib': (0, "8822efa61f2a385e0bc83ca5819d608111b2168a\trefs/heads/master")
+      'git ls-remote --heads /sw/MIRROR/root': (0, "87b87c4322d2a3fad315c919cb2e2dd73f2154dc\trefs/heads/master\nf7b336611753f1f4aaa94222b0d620748ae230c0\trefs/heads/v6-08-00-patches"),
```

Comment:
What will happen of the non `--offline` tests?

Reply:
In both offline and online mode, it will lookup the git heads locally.

Before the logic was:
1. Lookup git heads remotely (slow)
2. Decide, if we need to build
3. As part of the build update local repo (slow)
Cost: N git heads lookups + M repo updates, if build is needed
Best case: N (no build needed), worst case: 2N

Now it is:
1. Update local repos (slow)
2. Lookup git heads locally (fast)
3. Decide, if we need to build
Cost: N repo updates

Repo updates and remote git head lookups cost about the same in terms of wall time.

This PR is a regression in total wall time for online mode though, because there is no parallelization any more on the slow network requests that are executed on every build invocation. But as I said, I could easily add the parallelization, but there is an open issue about this.